src/ecs/commands.ts

```diff
@@ -0,0 +1,182 @@
+          )
+
+    if (await settings.isPromptEnabled(prompt)) {
+        const choice = await window.showWarningMessage(warningMessage, yes, yesDontAskAgain, no)
+        if (choice === undefined || choice === no) {
+            throw new CancellationError('user')
+        } else if (choice === yesDontAskAgain) {
+            settings.disablePrompt(prompt)
+        }
+    }
+
+    await service.toggleExecuteCommand()
+}
+
+export const runCommandInContainer = Commands.register('aws.ecs.runCommandInContainer', (obj?: unknown) => {
+    return telemetry.ecs_runExecuteCommand.run(async span => {
+        span.record({ ecsExecuteCommandType: 'command' })
+
+        const { container, task, command } = await runCommandWizard(obj)
+        const timeout = new Timeout(600000)
+        showMessageWithCancel('Running command...', timeout)
+
+        try {
+            const { path, args, dispose } = await container.prepareCommandForTask(command, task)
+            showOutputMessage(
+                `${moment().format(INSIGHTS_TIMESTAMP_FORMAT)}:  Container: "${
+                    container.description.name
+                }" Task ID: "${task}" Command: "${command}"`,
+                globals.outputChannel
+            )
+
+            const proc = new ChildProcess(path, args, { logging: 'noparams' })
+            await proc
+                .run({
+                    timeout,
+                    rejectOnError: true,
+                    rejectOnErrorCode: true,
+                    // TODO: `showOutputMessage` should not be writing to the logs...
+                    onStdout: text => {
+                        showOutputMessage(removeAnsi(text), globals.outputChannel)
+                    },
+                    onStderr: text => {
+                        showOutputMessage(removeAnsi(text), globals.outputChannel)
+                    },
+                })
+                .finally(dispose)
+        } catch (err) {
+            if (isUserCancelledError(err)) {
+                showOutputMessage('Cancelled command execution', globals.outputChannel)
+            }
+
+            const failedMessage = localize(
+                'AWS.ecs.runCommandInContainer.error',
+                'Failed to execute command in container.'
+            )
+            throw ToolkitError.chain(err, failedMessage)
+        } finally {
+            timeout.dispose()
+        }
+    })
+})
+
+// VSC is logging args to the PTY host log file if shell integration is enabled :(
+async function withoutShellIntegration<T>(cb: () => T | Promise<T>): Promise<T> {
+    const userValue = Settings.instance.get('terminal.integrated.shellIntegration.enabled', Boolean)
+
+    try {
+        await Settings.instance.update('terminal.integrated.shellIntegration.enabled', false)
+        return await cb()
+    } finally {
+        Settings.instance.update('terminal.integrated.shellIntegration.enabled', userValue)
+    }
+}
```

Comment:
This hack is my biggest concern right now as there's a slight risk of leaving the setting `false`. Without it, the session token gets logged by VS Code. Need to open an issue with them as they shouldn't be logging arguments from extensions.